src/navigator-card-body.jsx

```diff
@@ -0,0 +1,302 @@
+ */
+
+import React, { useCallback, useEffect, useMemo, useState, useRef } from "react";
+import {
+    Card, CardBody,
+    Flex,
+    Gallery,
+    Icon,
+    CardTitle, Spinner, CardHeader,
+} from "@patternfly/react-core";
+import { FileIcon, FolderIcon } from "@patternfly/react-icons";
+
+import cockpit from "cockpit";
+import { useDialogs } from "dialogs.jsx";
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+const compare = (sortBy) => {
+    const compareFileType = (a, b) => {
+        const aIsDir = (a.type === "dir" || a?.to === "dir");
+        const bIsDir = (b.type === "dir" || b?.to === "dir");
+
+        if (aIsDir && !bIsDir)
+            return -1;
+        if (!aIsDir && bIsDir)
+            return 1;
+        return 0;
+    };
+
+    switch (sortBy) {
+    case "az":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() < b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "za":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() > b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "last_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime > b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "first_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime < b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    default:
+        break;
+    }
+};
+
+export const NavigatorCardBody = ({
+    currentFilter,
+    files,
+    historyIndex,
+    isGrid,
+    path,
+    selected,
+    setHistory,
+    setHistoryIndex,
+    setSelected,
+    setSelectedContext,
+    sortBy,
+    loadingFiles
+}) => {
+    const [boxPerRow, setBoxPerRow] = useState(0);
+    const Dialogs = useDialogs();
+    const sortedFiles = useMemo(() => {
+        const compareFunc = compare(sortBy);
+
+        return files
+                .filter(file => {
+                    return file.name.toLowerCase().includes(currentFilter.toLowerCase());
+                })
+                .sort(compareFunc);
+    }, [files, currentFilter, sortBy]);
+    const isMounted = useRef(null);
+
+    function calculateBoxPerRow () {
+        const boxes = document.querySelectorAll(".item-button");
+        if (boxes.length > 1) {
+            let i = 0;
+            const total = boxes.length;
+            const firstOffset = boxes[0].offsetTop;
+            while (++i < total && boxes[i].offsetTop === firstOffset);
+            setBoxPerRow(i);
+        }
+    }
+
+    const onDoubleClickNavigate = useCallback((file) => {
+        const newPath = [...path, file.name].join("/");
+        if (file.type === "dir" || file.to === "dir") {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)